students/hyska_monika/lesson_08_regular_expressions/strong_password_detection.py

```diff
@@ -0,0 +1,55 @@
+"""
+Verify the password is strong
+rules:
+    min 8 characters length,
+    min 1 digits,
+    uppercase letter,
+    lowercase letter)
+"""
+import re
+
+
+def password_check(password):
+    length_error = len(password) < 8
+    uppercase_error = re.search(r"[A-Z]", password) is None
+    lowercase_error = re.search(r"[a-z]", password) is None
+    digit_error = re.search(r"\d", password) is None
+    password_ok = not (length_error or digit_error or
+                       uppercase_error or lowercase_error)
+    password_status = {'Password is strong.': password_ok,
```

Comment:
You could have started with the dictionary from the beginning:
```
password_status = dict(
    length=len(password) > 8,
    uppercase=re.search(...),
    ...
```
Then you could say, `password_ok = all(password_status)`. You could have messages as values or even exceptions in the other dictionary.

It's also a good idea to keep positive ideas as `True` in boolean variables so `uppercase` instead of `uppercase_error`. This is to avoid confusion when computing several levels of logic like: "not all statuses are different from false". It is better expressed as "any status is false".